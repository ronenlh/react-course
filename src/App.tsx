import React, { useState } from "react";
import styles from "./App.module.css";

export const sum = (a: number, b: number) => {
  return a + b;
};

export const foo = (str: string) => {
  return {
    name: str.toUpperCase() + '!'
  };
};

const App = () => {
  const [name, setName] = useState("");

  return (
    <div className={styles.contactUs}>
      <form
        onSubmit={(event) => {
          alert("A name was submitted: " + name);
          event.preventDefault();
        }}
      >
        <label>
          NAME: <em>*</em>
          <input
            type="text"
            name="name"
            value={name}
            onChange={(event) => setName(event.target?.value)}
          />
        </label>
        <input className={styles.submit} type="submit" value="Submit" />
      </form>
    </div>
  );
};

export default App;
