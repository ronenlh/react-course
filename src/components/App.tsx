import Counter from './Counter';
import React from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { CounterState } from '../reducers';

const App: React.FC = () => {
    const count = useSelector<CounterState, number>((state) => state.counter);
    const dispatch = useDispatch()

    return (<Counter
        value={count}
        onIncrement={() => dispatch({ type: 'INCREMENT' })}
        onDecrement={() => dispatch({ type: 'DECREMENT' })}
    />);
};

export default App;
