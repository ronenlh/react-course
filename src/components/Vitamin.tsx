import React from 'react';

const Vitamin = () => {
    return <h1>Vitamin Page!!!!</h1>
}
 
export default Vitamin;
