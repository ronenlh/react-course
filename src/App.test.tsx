import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import App from './App';
import Vitamin from './components/Vitamin';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Vitamin renders successfully', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter initialEntries={[ '/vitamin']}>
      <Vitamin />
    </MemoryRouter>
    , div);
  ReactDOM.unmountComponentAtNode(div);
});