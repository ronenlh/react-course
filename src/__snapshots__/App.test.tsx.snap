// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`learning tests tests snap 1`] = `
Object {
  "name": "HELLO WORLD!",
}
`;
